sources/packages/backend/apps/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -177,6 +181,14 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
         ASSESSMENT_INVALID_OPERATION_IN_THE_CURRENT_STATE,
       );
     }
+
+    if (assessment.application.currentAssessment.id !== assessment.id) {
+      throw new CustomNamedError(
+        `An assessment other than the current one may not be approved.`,
```

Comment:
Please update the `@ApiUnprocessableEntityResponse` in the `confirmAssessmentNOA` controller method.
```ts
@ApiUnprocessableEntityResponse({
    description: "Student not found or assessment confirmation failed or an assessment other than the current one may not be approved.",
  })
```

Reply:
Updated